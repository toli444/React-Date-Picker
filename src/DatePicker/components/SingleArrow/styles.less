.double-arrow {
    border-top: 2px solid #2A2A2A;
    border-bottom: 2px solid #2A2A2A;
    cursor: pointer;
    padding: 3px;
}
