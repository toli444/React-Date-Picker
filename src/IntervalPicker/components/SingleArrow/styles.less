@hover-color: #dadada;

.single-arrow {
  cursor: pointer;
  padding: 3px;

  &:hover {
    background-color: #dadada;

    &:active {
      background-color: darken(@hover-color, 15%);
    }
  }
}
