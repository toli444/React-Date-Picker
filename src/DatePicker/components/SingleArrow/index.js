import React from 'react';
import styles from './styles.css';

export default function SingleArrow(props) {
  return <div className={styles['double-arrow']}>{props.left ? '<' : '>'}</div>;
};

