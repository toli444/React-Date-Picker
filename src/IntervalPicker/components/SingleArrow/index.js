import React from 'react';
import styles from './styles.less';
import PropTypes from 'prop-types';

const SingleArrow = ({left, onClick}) => {
  return <div onClick={onClick} className={styles['single-arrow']}>{left ? '<' : '>'}</div>;
};

SingleArrow.propTypes = {
  left: PropTypes.bool,
  onClick: PropTypes.func,
};

export default SingleArrow;



