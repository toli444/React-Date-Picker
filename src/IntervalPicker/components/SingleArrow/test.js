import React from 'react';
import SingleArrow from '../SingleArrow';
import { shallow } from 'enzyme';

describe('SingleArrow', () => {
  it('renders correctly', () => {
    const wrapper  = shallow(<SingleArrow />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders left arrow if left', () => {
    const wrapper = shallow(<SingleArrow left/>);

    expect(wrapper.get(0).props.children).toEqual('<');
  });

  it('renders right arrow otherwise', () => {
    const wrapper = shallow(<SingleArrow />);

    expect(wrapper.get(0).props.children).toEqual('>');
  });

  it ('calls correct function when click', () => {
    const onClickMock = jest.fn();
    const wrapper = shallow(
      <SingleArrow onClick={onClickMock}/>,
    );

    wrapper.simulate('click');

    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
