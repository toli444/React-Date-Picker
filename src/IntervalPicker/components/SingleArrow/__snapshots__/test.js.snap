// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleArrow renders correctly 1`] = `
<div
  className="single-arrow"
>
  &gt;
</div>
`;
